pkg/instancetype/apply/vmi_test.go

```diff
@@ -0,0 +1,1364 @@
+package apply_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	"kubevirt.io/client-go/api"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+
+	v1 "kubevirt.io/api/core/v1"
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+
+	instancetypeapply "kubevirt.io/kubevirt/pkg/instancetype/apply"
+)
+
+var _ = Describe("Apply to vmi", func() {
+	var (
+		vmi              *v1.VirtualMachineInstance
+		instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec
+		preferenceSpec   *instancetypev1beta1.VirtualMachinePreferenceSpec
+		field            *k8sfield.Path
+	)
+
+	BeforeEach(func() {
+		vmi = api.NewMinimalVMI("testvmi")
+
+		vmi.Spec = v1.VirtualMachineInstanceSpec{
+			Domain: v1.DomainSpec{},
+		}
+		field = k8sfield.NewPath("spec", "template", "spec")
+	})
+
+	Context("instancetype.spec.NodeSelector", func() {
```

Comment:
Can we break these individual parts out alongside their implementations, so in this case with `nodeselector_tests.go` using:

https://github.com/kubevirt/kubevirt/blob/c31953808e2521505a175c788b95d72b0ad707bd/pkg/instancetype/apply/nodeselector.go